wsi/drm/drmdisplay.cpp

```diff
@@ -566,6 +661,85 @@ void DrmDisplay::GetDisplayCapabilities(uint32_t *numCapabilities,
   }
 }
 
+bool DrmDisplay::GetHdrCapabilities(uint32_t *outNumTypes, int32_t *outTypes,
+                                    float *outMaxLuminance,
+                                    float *outMaxAverageLuminance,
+                                    float *outMinLuminance) {
+  if (NULL == outNumTypes) {
+    ALOGE("outNumTypes couldn't be NULL!");
+    return false;
+  }
+
+  if (NULL == outMaxLuminance) {
+    ALOGE("outMaxLuminance couldn't be NULL!");
+    return false;
+  }
+
+  if (NULL == outMaxAverageLuminance) {
+    ALOGE("outMaxAverageLuminance couldn't be NULL!");
+    return false;
+  }
+
+  if (NULL == outMinLuminance) {
+    ALOGE("outMinLuminance couldn't be NULL!");
+    return false;
+  }
+
+  if (display_hdrMd) {
+    // HDR meta block bit 3 of byte 3: STPTE ST 2084
+    if (display_hdrMd->eotf & 0x04) {
+      if(outTypes != NULL)
+          *(outTypes + *outNumTypes) = (uint32_t)EOTF_ST2084;
+      (*outNumTypes)++;
+    }
+    // HDR meta block bit 4 of byte 3: HLG
+    if (display_hdrMd->eotf & 0x08) {
+      if(outTypes != NULL)
+        *(outTypes + *outNumTypes) = (uint32_t)EOTF_HLG;
+      (*outNumTypes)++;
+    }
+    double outmaxluminance, outmaxaverageluminance, outminluminance;
+    // Luminance value = 50 * POW(2, coded value / 32)
+    // Desired Content Min Luminance = Desired Content Max Luminance * POW(2,
+    // coded value/255) / 100
+    outmaxluminance = pow(2.0, display_hdrMd->desired_max_ll / 32.0) * 50.0;
+    *outMaxLuminance = float(outmaxluminance);
+    outmaxaverageluminance =
+        pow(2.0, display_hdrMd->desired_max_fall / 32.0) * 50.0;
+    *outMaxAverageLuminance = float(outmaxaverageluminance);
+    outminluminance = display_hdrMd->desired_max_ll *
+                      pow(2.0, display_hdrMd->desired_min_ll / 255.0) / 100;
+    *outMinLuminance = float(outminluminance);
+  }
+
+  return true;
+}
+
+bool DrmDisplay::GetPerFrameMetadataKeys(uint32_t *outNumKeys,
+                                         int32_t *outKeys) {
+  *outNumKeys = KEY_NUM_PER_FRAME_METADATA_KEYS;
```

Comment:
KEY_NUM_PER_FRAME_METADATA_KEYS is  a enum, and not assigned with a value explicitly